### PR TITLE
EIP-6110: Queue deposit requests and apply them during epoch processing

### PR DESCRIPTION
**Pull request by lucassaldanha**
## PR Description
- Updated block processing and epoch processing with changes from https://github.com/ethereum/consensus-specs/pull/3818
- Updated reference tests to v1.5.0-alpha.7
  - Had to ignore a few tests:
    - ` epoch_processing/process_consolidations` and `operations/consolidation_request` (will be re-enabled as part of https://github.com/Consensys/teku/issues/8617)
    - 3 sanity/blocks tests realted to EL withdrawals that are being investivated, and other CL teams are also having issues with (https://github.com/Consensys/teku/issues/8680)

Pending:
- Remove duplication on deposit related code between Epoch Processing and Block Processing. Better done as a separate PR to avoid adding many changes on this PR.

## Fixed Issue(s)
part of #8618 

## Documentation

- [x] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/versions/electra/statetransition/epoch/EpochProcessorElectra.java [hunk 1]
```diff
@@ -186,73 +201,176 @@ public void processEffectiveBalanceUpdates(
...
-        stateElectra.getPendingBalanceDeposits();
-    for (final PendingBalanceDeposit deposit : pendingBalanceDeposits) {
-      final Validator validator = state.getValidators().get(deposit.getIndex());
+      // Check if number of processed deposits has not reached the limit, otherwise, stop processing
+      if (nextDepositIndex
+          >= SpecConfigElectra.required(specConfig).getMaxPendingDepositsPerEpoch()) {
+        break;
+      }
+
+      // TODO-lucas What is the impact of doing this? Should we use the transition cache?
```
Comment (StefanBratanov) [inline ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/versions/electra/statetransition/epoch/EpochProcessorElectra.java]: Hmm, yeah this for sure needs to change. Think we should use `ValidatorsUtil.getValidatorIndex()` (internally uses the cache) and then just use .get(index) on the validators in the state.

**Review by StefanBratanov — APPROVED** [review]
LGTM, logic wise with one comment on necessary optimization

File: ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/versions/electra/statetransition/epoch/EpochProcessorElectra.java (hunk 1 above)
Comment (lucassaldanha) [reply ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/versions/electra/statetransition/epoch/EpochProcessorElectra.java]: Done!

File: ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/common/statetransition/epoch/AbstractEpochProcessor.java [hunk 2]
```diff
@@ -122,7 +122,7 @@ protected void processEpoch(final BeaconState preState, final MutableBeaconState
     processRegistryUpdates(state, validatorStatuses.getStatuses());
     processSlashings(state, validatorStatuses);
     processEth1DataReset(state);
-    processPendingBalanceDeposits(state);
+    processPendingDeposits(state);
```
Comment (tbenr) [inline ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/common/statetransition/epoch/AbstractEpochProcessor.java:135]: I thought about the mutating `ValidatorStatuses` and seems an additional complication to me.
At current stage I'd just approach the "problem" with verbose comments.
We could be noisy here saying that here `processPendingDeposits` may add validators in the state and `ValidatorStatuses` won't contain these new added vals.

**Review by tbenr — APPROVED** [review]
LGTM

File: ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/common/statetransition/epoch/AbstractEpochProcessor.java (hunk 2 above)
Comment (lucassaldanha) [reply ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/common/statetransition/epoch/AbstractEpochProcessor.java:135]: Added a big as warning in AbstractEpochProcessor :)

**Pull request merged**